### PR TITLE
RUCK-12 Create anomaly detection model 

### PR DESCRIPTION
**Pull request by erf300three**
We have create an anomaly detection model which learns from weakly labelled video clips to determine if a video clip contains an anomaly (a tackle) and what location that tackle occurred in. This model has been trained with the same videos that we have used in our action recognition model though labels have only been applied at the video level rather than the clip level. 

The loss function and the architecture for this model were taken from the following paper: https://openaccess.thecvf.com/content_cvpr_2018/papers/Sultani_Real-World_Anomaly_Detection_CVPR_2018_paper.pdf

We have not yet evaluated the performance of this model though it does not look promising as the loss value did not move much across the many epochs it trained for suggesting that it did not learn the key features of a tackle keeping its predictions around 0.5 for most inputs. 

More evaluation and testing will need to be put into this model to determine if this is a suitable replacement to the other model and determining if these architecture of model did in fact work or not.

### CODE REVIEW

**Comment by erf300three** [comment]
As mentioned in the PR we need too write evaluation functions for this model and get it running on the test set to see how well it does predicting unseen data. We also might want to look for a different validation metric than just looking at the average loss across the validation set.  This did teach me a lot about how to work with DNNs within pytorch which I will be able to apply to the other models and work that I need to complete.
